### DISCUSSION TITLE
Interface vs Types [Typescript]

### DISCUSSION THREAD

**Discussion by somratAtSekaiLab** [monstar-lab-oss/admin-panel-template-reactjs#88] [General] [answered]
- [ ] What to use
- [ ] Where to use
- [ ] Why to use

**Comment by yukinoda** [comment]
`interface` can be extended and `type` not
https://www.educba.com/typescript-type-vs-interface/

This may one good way when thinking of using which one 🤔

**Reply by geekonthemove** [reply]
Just for documentation's sake, as mentioned in the meeting, `type` can also be "extended" using the `&` operator.

**Comment by somratAtSekaiLab** [comment]
Nice article discussing interface vs types
https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c

**Comment by yukinoda** [comment]
info: `OpenApi Generator` generates `interface`

**Comment by themitvp** [comment, marked as answer]
I suggest: 
- interface for components props
- types for everything else

so as an example:
```tsx
// could be defined in a file like `types/user.types.ts`
type UserDef = {
   name: string;
}

// in a file like `UserProfile.tsx`
interface UserProfileProps {
   user: UserDef;
}
const UserProfile = ({ user }: UserProfileProps) => {
   return <div>{user.name}</div>;
}
```


Reason, other than it being a personal preference, I think it makes it easy to read if we want to extend our component props, for example:
```tsx
interface UserProfileProps extends CardProps {
```

**Reply by yukinoda** [reply]
@somratAtSekaiLab @jacksonjp @geekonthemove 
Could we mark this as the answer?
